source/includes/api/_vocabulary.md

```diff
@@ -1,694 +1,1297 @@
+
+- `tags`: an array of tags
+- `application`: the application associated with the vocabulary.
+
+You can create different vocabularies for different applications in a single request.
+
+Assuming the creation was successful, the response will contain a list of all vocabularies and their respective tags
+associated with the specified resource, for all applications.
+
+If you want to create new vocabulary/tags for your resources, you must have the necessary user account permissions.
+Specifically:
+
+- the user must be logged in and belong to the same application as the resource
+- the user must match one of the following:
+    - have role `ADMIN`
+    - have role `MANAGER` and be the resource's owner (through the `userId` field of the resource)
+
+When creating vocabularies/tags for a resource, the dataset id specified in the URL is validated, and the requests fail
+if a dataset with the given id does not exist. When creating a vocabulary for a widget or layer, you should ensure you
+specify the dataset id that matches that of the widget/layer, as that validation is not done automatically - this is a
+known limitation of the current implementation, and may be modified at any time, and invalid resources (those where the
+layer's/widget's dataset id does not match the dataset id defined in the resource) may be purged without prior warning.
+
+#### Errors for creating a single vocabulary/tags for a resource
+
+Error code     | Error message  | Description
+-------------- | -------------- | --------------
+400            | This relationship already exists | The resource already has one or more vocabularies with the same name(s) for the specified application(s)
+400            | - `<value>`: `<value>` check failed. -  | The body object has a `<value>` key which content is invalid (is not an object or is missing `tags` and/or `application`).
+401            | Unauthorized   | You are not authenticated.
+403            | Forbidden      | You are trying to create a vocabulary for a resource whose `application` value is not associated with your user account.
+404            | 404 - {\"errors\":[{\"status\":404,\"detail\":\"Dataset with id `<dataset id>` doesn't exist\"}]}      | You are trying to create a vocabulary for a dataset that doesn't exist.
+
+## Updating vocabularies/tags
 
-There is also an endpoint that allows you to create multiple relationships in the same request.
+If you have already associated vocabularies and tags to your resources and would like to modify those tags, the next set
+of endpoints is for you.
 
-> Creating multiple relationships between a Vocabulary and a Dataset
+There are three types of endpoints for updating existing vocabulary/tags:
+
+- endpoints that allow you to modify a single vocabulary (and its tags) for a resource
+- endpoints that allow you to modify multiple vocabularies/tags for a single resource with a single request
+- endpoint that allow you to add new tags to an existing dataset vocabulary
```

Comment:
```suggestion
- endpoint that allows you to add new tags to an existing vocabulary. This endpoint only exists for dataset resources, not widgets or layers.
```
Just thought it would be good to be more explicit here, since the original wording caught me off guard.